src/main.cpp

```diff
@@ -4477,6 +4477,15 @@ bool static LoadBlockIndexDB()
     fSpentIndex = fInsightExplorer;
     fTimestampIndex = fInsightExplorer;
 
+    bool fLightWalletd = false;
+    pblocktree->ReadFlag("lightwalletd", fLightWalletd);
+    LogPrintf("%s: light wallet daemon %s\n", __func__, fLightWalletd ? "enabled" : "disabled");
+    if (fLightWalletd) {
+        fAddressIndex = true;
+        fSpentIndex = false;
+        fTimestampIndex = false;
+    }
```

Comment:
If both `-insightexplorer` and `-lightwalletd` are set, this causes `-insightexplorer` to be ignored. We should either make them mutually exclusive (only one can be set, enforced by `init.cpp`), or handle the case here where both are set. I'm weakly leaning towards making them mutually exclusive.

Reply:
Can't this be solved by just setting `fAddressIndex = true;` here? It's not logically inconsistent to want to support both the Insight Explorer and lightwalletd from the same zcashd.

Reply:
Went with @daira suggestion at https://github.com/zcash/zcash/pull/4402/commits/7aed10f195c961408e6c4bbbae2595c2c17e5a89